planner/core/find_best_task.go

```diff
@@ -547,7 +547,7 @@ func (ds *DataSource) skylinePruning(prop *property.PhysicalProperty) []*candida
 		}
 		var currentCandidate *candidatePath
 		if path.IsTablePath() {
-			if path.StoreType == kv.TiFlash && prop.TaskTp != property.CopTiFlashGlobalReadTaskType {
+			if path.StoreType == kv.TiFlash {
```

Comment:
I cannot see any difference with these two branches. It seems only invalid condition is prop requires flash prop but the candidate path is tikv only.